src/App.tsx

```diff
@@ -15,48 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  length = 'length',
+  alphabet = 'alphabet',
+  noSort = '',
+}
+
+function sortGoods(sortField: SortType, isReversed: boolean) {
+  const goods = [...goodsFromServer];
+
+  if (sortField) {
+    goods.sort((a, b) => {
+      switch (sortField) {
+        case SortType.length:
+          return a.length - b.length;
+        case SortType.alphabet:
+          return a.localeCompare(b);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? goods.reverse() : goods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.noSort);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goods = sortGoods(sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-info',
+            { 'is-light': sortField !== SortType.alphabet })}
+          onClick={() => setSortField(SortType.alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button is-success',
+            { 'is-light': sortField !== SortType.length })}
+          onClick={() => setSortField(SortType.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField !== '' || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsReversed(false);
+              setSortField(SortType.noSort);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {goods.map((good, _, arr) => (
+          <li
+            data-cy="Good"
+            key={`${good} from ${arr}`}
```

Comment:
Having just `good` for the key is okay here
In any case the part "from ${arr}" will not be unique for the goods, so you can omit it and make all mapping simpler.